app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,23 @@
 class PostsController < ApplicationController
   def index; end
 
-  def show; end
+  def show
+    @post = Post.where(id: params[:id])[0]
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    author = User.find(params['post']['author_id'].to_i)
```

Comment:
- [ ] A post is created on behalf of the currently logged-in user (current_user) of the app. Remember the `current_user` method you have in your `application_controller`? that's what we're going to use to determine the current user when creating a post.
See suggestion below:

```ruby
 @post = @current_user.posts.new(post_params)
```